### PR TITLE
USART1 connects to PCLK1.

### PR DESCRIPTION
**Pull request by richard7770**
The clock tree figures of RM0366, RM0365, RM0364, RM0316, and RM0313 show that the default setting of RCC::CFGR3::USART1SW will, for some variants, make the port use PCLK1 for reference, and not PCLK2 where port is resident.

The affected variants are stm32f3xxx4/6/8, except stm32f37xxx.

This is consistent with my observations on stm32f344r8, which fails to produce the correct baud rate:
- If sysclk is configured above 32 MHz and both pclk are untouched (pclk1 is then divided, but pclk2 is not), or
- If the pclk's are specifically configured differently.

This might be related to #109, although I don't see why the discussed code snippet should cause different pclk's.

### CODE REVIEW

**Comment by Sh3Rm4n** [comment]
> This is consistent with my observations on stm32f344r8, which fails to produce the correct baud rate:
> 
> * If sysclk is configured above 32 MHz and both pclk are untouched (pclk1 is then divided, but pclk2 is not), or
> * If the pclk's are specifically configured differently.

Thanks! I'll take a deeper look later today :)

File: src/serial.rs [hunk 1]
```diff
@@ -869,8 +869,17 @@ macro_rules! usart_var_clock {
 }
 
 cfg_if::cfg_if! {
-    if #[cfg(any(feature = "svd-f301", feature = "svd-f3x4"))] {
-        usart_var_clock!([(1,2)]);
+    if #[cfg(all(
+        any(
+            feature = "mem-4",
+            feature = "mem-6",
+            feature = "mem-8",
```
Comment (Sh3Rm4n) [inline src/serial.rs:876]: Is this feature constraint not describable via the `svd-*` features? I mean the `mem-*` features do work, but are rather intended to distinguish memory differences and this seems not related.

Comment (richard7770) [reply src/serial.rs:876]: I used them because, for example f302x6/8 are affected, while f302xb/c/d/e are not.

Comment (Sh3Rm4n) [reply src/serial.rs:876]: Hm. Yeah, this is not representable with the `svd-*`-features. But I would suggest to used `stm32f303xb`, `stm32f303xc` ... so that it is more obvious, when reading it :)

Comment (Sh3Rm4n) [reply src/serial.rs:876]: > (Fig 12+13 vs. Fig 14 of RM0365)

Thanks for that info! I can see the difference. So I think the implementation is fine. Will still look at it later today :)

**Comment by Sh3Rm4n** [comment]
As this is a bugfix effecting `v0.7` would you like to add a CHANGELOG entry?

**Comment by richard7770** [comment]
I didn't mention #109 in the changelog, because I'm unsure whether this is a fix to that bug.

**Review by Sh3Rm4n — APPROVED** [review]

**Comment by Sh3Rm4n** [comment]
> I didn't mention #109 in the changelog, because I'm unsure whether this is a fix to that bug.

That's okay. I'll have to invest a bit more time on that issue anyway, and as it might not be related, let's ignore it for now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
   features, can no longer be disabled. ([#259])
   1. The "unproven" features are no longer unproven and used anywhere anyways.
   2. This crate was not building successfully without the unproven feature.
+- Set the correct baud rate for chips where `USART1SW_A::PCLK` leads to a baud rate 
+  derived from PCLK1, rather than the ports own bus clock, PCLK2. ([#260])
```
Comment (Sh3Rm4n) [inline CHANGELOG.md:36]:
```suggestion
- Set the correct baud rate for chips where `USART1SW_A::PCLK` leads to a
  baud rate derived from PCLK1, rather than the ports own bus clock, PCLK2. 
  ([#260])
```

Markdown linter is unhappy :( https://github.com/stm32-rs/stm32f3xx-hal/pull/260/checks?check_run_id=3182875827

**Comment by Sh3Rm4n** [comment]
> If sysclk is configured above 32 MHz and both pclk are untouched (pclk1 is then divided, but pclk2 is not), or

That reminds me, that I probably should change the processor clock to 32 MHz or above instead of the default 8 Mhz for the [uart tests](https://github.com/stm32-rs/stm32f3xx-hal/blob/ee27f328639c3c75c6f5ab61be06dbccea883df4/testsuite/tests/uart.rs#L64)

**Comment by Sh3Rm4n** [comment]
I have not run a local test to verify this behavior nor check the erroneous old behavior, but I'll trust you and the datasheet on that one. 

Thanks for the catch and this contribution. Much appreciated! :+1:
